### PR TITLE
IP Grade

### PR DESCRIPTION
**Pull request by skorpiom**
Hey,

just added grade endpoints 
really sorry for the mess, pls ignore files with the name different than Grade - (ex. Auth and Middlewares)

### CODE REVIEW

File: Api/Src/Services/GradeService.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as mongoose from "mongoose";
+
+import GradeRepository from "../Repositories/GradeRepository";
+import { GradeType } from "../Models/Grade";
+
+export default class CourseService {
```
Comment (kjreczek) [inline Api/Src/Services/GradeService.ts:6]: CourseService? 🙈

File: Api/Test/GradeService.test.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    constructor() {
+        super(Grade);
+    }
+
+    async getAll(){
+        return this.grades;
+    }
+
+    async getById(id: Types.ObjectId){
+        return this.grades.find(course => course._id === id);
```
Comment (kjreczek) [inline Api/Test/GradeService.test.ts:22]: Maybe grade instead of course :P

File: Api/Src/Services/GradeService.ts (hunk 1 above)
Comment (skorpiom) [reply Api/Src/Services/GradeService.ts]: haha spotted, I inspired  myself with some clean and smart solutions

**Review by kjreczek — APPROVED** [review]

**Review by fszwedo — APPROVED** [review]
all seems fine, haven't found any code issues - endpoints work fine, just had to create a dummy section model first to test (took me a while to figure out what's going on) :)

try-catches are probably obsolete once the id validation and all the error handling will be in place but they definitely don't hurt

**Comment by fszwedo** [comment]
@skorpiom please check the conflicts and merge once you'll finish :)

**Comment by skorpiom** [comment]
@fszwedo sorry i should have mentioned about section model it would have saved your time but it'll be work fine when merged

**Pull request merged**